decidim-bulletin_board-app/app/commands/open_ballot_box.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+# A command with all the business logic to open the election's ballot box
+class OpenBallotBox < Rectify::Command
+  include LogEntryCommand
+
+  # Public: Initializes the command.
+  #
+  # authority - The authority sender of the vote message
+  # message_id - The message identifier
+  # signed_data - The signed message received
+  def initialize(authority, message_id, signed_data)
+    @client = @authority = authority
+    @message_id = message_id
+    @signed_data = signed_data
+  end
+
+  # Executes the command. Broadcasts these events:
+  #
+  # - :ok when everything is valid.
+  # - :invalid if the received data wasn't valid.
+  #
+  # Returns nothing.
+  def call
+    return broadcast(:invalid, error) unless
+      valid_log_entry?("open_ballot_box")
+
+    election.with_lock do
+      return broadcast(:invalid, error) unless
+        valid_step?(election.ready?) &&
+        valid_client?(authority.authority?) &&
+        valid_author?(message_identifier.from_authority?)
```

Comment:
I think we should add a `&& process_message` call here, to allow voting schemes to be informed about this messages reception (also for the close_ballot_box message). We don't need to modify the dummy scheme and it won't do anything with the message, but the EG wrapper will use it.

Reply:
I was in doubt if the `process_message` was needed here, in any case I'll add it.